### ISSUE TITLE
AttributeError: 'Dataframe' object has no attribute 'input'

### ISSUE THREAD

**Issue by baiqunicui** [chrisgoringe/Styles-Editor#20]
```
Error executing callback ui_tabs_callback for C:\Users\urfan\stable-diffusion-webui\extensions\Styles-Editor\scripts\main.py
Traceback (most recent call last):
  File "C:\Users\xxx\stable-diffusion-webui\modules\script_callbacks.py", line 134, in ui_tabs_callback
    res += c.callback() or []
  File "C:\Users\xxx\stable-diffusion-webui\extensions\Styles-Editor\scripts\main.py", line 216, in on_ui_tabs
    cls.dataeditor.input(fn=cls.save_styles, inputs=[cls.dataeditor, cls.autosort_checkbox], outputs=cls.dataeditor)
AttributeError: 'Dataframe' object has no attribute 'input'
```

there is an error in this line and can't run this extension, thank you.

**Comment by chrisgoringe** [comment]
Could you post your styles.csv file? I'd like to try to reproduce your error.

**Comment by baiqunicui** [comment]
my styles.csv inside webui main folder
[styles.csv](https://github.com/chrisgoringe/Styles-Editor/files/11711994/styles.csv)

and i can't see ant style editor tab in my ui
![Screenshot 2023-06-10 at 15-32-58 Stable Diffusion](https://github.com/chrisgoringe/Styles-Editor/assets/67366810/e4b18784-f0c0-4e2d-bf6c-3b871457837a)

thank you

**Comment by chrisgoringe** [comment]
OK, the version of gradio you're running is too old. I'll update the extension to do the install. Assuming a normal installation, you can do this at the command line:
```
venv/Scripts/activate.bat
pip install gradio==3.31.0 --upgrade
```
3.31.0 is the version used by the latest A1111 update, so you could also just update your base.

**Comment by chrisgoringe** [comment]
Will be fixed as #23
